+++ gcl-2.6.8/gcl-tk/tkMain.c.orig	2009-08-14 20:02:00.000000000 -0300
--- gcl-2.6.8/gcl-tk/tkMain.c	2009-08-14 20:03:41.000000000 -0300
@@ -34,9 +34,10 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
+#define USE_INTERP_RESULT
 #include <tcl.h>
 #include <tk.h>
+#undef USE_INTERP_RESULT
-
 
 
 #if (TK_MINOR_VERSION==0 && TK_MAJOR_VERSION==4)
@@ -121,7 +122,7 @@
 TCL_signal_error(x)
      char *x;
 {char buf[300] ;
+ snprintf(buf, sizeof(buf), "error %s",x);
- sprintf("error %s",x);
  Tcl_Eval(interp,buf);
  dfprintf(stderr,x);
 }
