diff -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2026-08-02 18:29:55.590659459 +0200
+++ b/Makefile.am	2026-08-02 18:29:55.969468010 +0200
@@ -245,7 +245,7 @@
 		mkdir h && \
 		ln -snf ../../h/cmpinclude.h h/ && \
 		GCL_LSPSYSDIR=../$(srcdir)/unixport/ \
-			../unixport/raw_$* $$(dirname $$(pwd))/unixport/ -libdir $$(dirname $$(pwd))/ \
+			setarch -RX ../unixport/raw_$* $$(dirname $$(pwd))/unixport/ -libdir $$(dirname $$(pwd))/ \
 			    < <(cat ../unixport/init_raw.lsp <(echo "(system:save-system \"../$@\")")) && \
 		rm -f ../unixport/gcl_cmpnopt.lsp
 	rm -rf sb_$*
@@ -260,7 +260,7 @@
 	touch $@
 unixport/gcl_cmpnopt_%.lsp: unixport/lib%.a | unixport/%
 	echo "(mapc (quote load) (directory \"$*/*.hsh\"))" \
-	     "(compiler::dump-inl-hash \"$@\")" | $|
+	     "(compiler::dump-inl-hash \"$@\")" | setarch -RX $|
 
 $(addprefix unixport/lib,$(addsuffix .a,pre_gcl $(MY_DIRS))): \
 unixport/lib%.a: lib/libbase_gcl.a $(LIBGPROF) unixport/sys_%.o
@@ -377,14 +377,14 @@
 
 
 %/c1: | unixport/%
-	echo "(pcl::compile-pcl)" | $|
+	echo "(pcl::compile-pcl)" | setarch -RX $|
 	touch $@
 
 %/sys-package.lisp: %/c1 | unixport/%
 	echo "(pcl::load-pcl)" \
 	     "(compiler::get-packages-ansi \
 	         (quote (:walker :iterate :pcl :slot-accessor-name)) \
-	         \"$@\")" | $|
+	         \"$@\")" | setarch -RX $|
 
 %/sys-proclaim.lisp: %/c1 | unixport/%
 	echo "(pcl::load-pcl)" \
@@ -396,7 +396,7 @@
 			\"*.*p\" \
 			(make-pathname \
 			  :directory (pathname-directory \
-				        (si::file (quote pcl::renew-sys-files))))))))" | $|
+				        (si::file (quote pcl::renew-sys-files))))))))" | setarch -RX $|
 %/p1.lisp: | unixport/%
 	echo "(in-package :si)" \
 	     "(export (quote %structure-name))" \
@@ -408,11 +408,11 @@
 	rm -rf $*/*.o
 	echo "(mapc (quote load) (list $(patsubst %,\"%\",$^)))" \
 	     "(pcl::compile-pcl)" \
-	     "(compiler::dump-inl-hash \"$@\")" | $|
+	     "(compiler::dump-inl-hash \"$@\")" | setarch -RX $|
 
 %/all.hsh: %/p1.lisp %/all.hs1 | unixport/pcl_gcl
 	echo "pcl conflicts:"
-	echo "(pcl::load-pcl)(si::all-conflicts)" | $|
+	echo "(pcl::load-pcl)(si::all-conflicts)" | setarch -RX $|
 	cat $^ > $@
 
 $(addprefix pcl_gcl/,$(PCL_OBJS)): pcl_gcl/all.hsh 
@@ -472,7 +472,7 @@
 # parallel builds can only have one target accessing an intermediate file
 # solved with BUILT_SOURCES
 o/%.ini:  o/%.c | o/grab_defs
-	@$(CPP) $(AM_CPPFLAGS) -DINICOMP -DNO_DEFUN $(CPPFLAGS) $< | $| > new_$(@F)
+	@$(CPP) $(AM_CPPFLAGS) -DINICOMP -DNO_DEFUN $(CPPFLAGS) $< | setarch -RX $| > new_$(@F)
 	@([ -e $@ ] && cmp new_$(@F) $@) || mv -v new_$(@F) $@
 	@rm -f new_$(@F)
 
@@ -485,7 +485,7 @@
 
 sb_ansi-tests/test_results: ansi-tests | unixport/saved_ansi_gcl
 	[ -d $(@D) ] || (mkdir $(@D) && cp $</*.lsp $(@D)) # FIXME testsuite in separate directory, hard
-	cd $(@D) && echo '(load "gclload")' | ../$| |tee $(@F)
+	cd $(@D) && echo '(load "gclload")' | setarch -RX ../$| |tee $(@F)
 
 BENCH=boyer browse ctak dderiv deriv destru destru-mod div2 fft fft-mod fprint fread frpoly \
       puzzle puzzle-mod stak tak takl tak-mod takr tprint traverse triang triang-mod
@@ -510,15 +510,15 @@
 	     "(proclaim-file \"$<\")" \
 	     "(compile-file \"$<\" :output-file \"$(@D)/$*.o\")" \
 	     "(load \"$(<D)/test-help.lsp\")" \
-	     "(do-test \"$(@D)/$*\" \"$(@D)/$*.tim\")" | $(word 1,$|)
+	     "(do-test \"$(@D)/$*\" \"$(@D)/$*.tim\")" | setarch -RX $(word 1,$|)
 
 gcl-tk/%.o: gcl-tk/%.lisp | unixport/saved_ansi_gcl
 	[ -d $(@D) ] || mkdir -p $(@D)
-	$(word 1,$|) -compile $< -o $@
+	setarch -RX $(word 1,$|) -compile $< -o $@
 
 gcl-tk/%.o: gcl-tk/%.lsp | unixport/saved_ansi_gcl
 	[ -d $(@D) ] || mkdir -p $(@D)
-	$(word 1,$|) -compile $< -o $@
+	setarch -RX $(word 1,$|) -compile $< -o $@
 
 CLEANFILES+=gcl-tk/demos/index.lsp gcl-tk/demos/gc-monitor.o
 gcl-tk/demos/index.lsp: gcl-tk/tkl.o | unixport/saved_gcl
@@ -527,7 +527,7 @@
 
 sb_cmpnew/gcl_collectfn.o: cmpnew/gcl_collectfn.lsp | unixport/saved_gcl
 	[ -d $(@D) ] || mkdir -p $(@D)
-	$(word 1,$|) -compile $< -o $@
+	setarch -RX $(word 1,$|) -compile $< -o $@
 
 # Keep generated .c from .d for debuginfo (do not delete after compile)
 %.c: %.d | bin/dpp
diff -urN a/bin/gcl.in b/bin/gcl.in
--- a/bin/gcl.in	2025-04-10 20:21:26.150034344 +0200
+++ b/bin/gcl.in	2026-08-02 18:29:55.969126925 +0200
@@ -26,7 +26,7 @@
 fi
 SYS=$DIR/unixport
 
-exec $SYS/$EXE -dir $SYS/ -libdir $DIR/ \
+exec setarch -RX $SYS/$EXE -dir $SYS/ -libdir $DIR/ \
    -eval '(setq si::*allow-gzipped-file* t)' \
    -eval '(setq si::*tk-library* "@TK_CONFIG_PREFIX@")' \
    -eval "(setq si::*default-info-files* (list \"gcl$EXT-si.info\" \"gcl$EXT-tk.info\" \"gcl$EXT-dwdoc.info\" \"gcl$EXT.info\"))" \
