diff -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2025-04-11 06:57:43.582839865 +0200
+++ b/Makefile.am	2026-08-02 18:29:55.590659459 +0200
@@ -529,6 +529,7 @@
 	[ -d $(@D) ] || mkdir -p $(@D)
 	$(word 1,$|) -compile $< -o $@
 
+# Keep generated .c from .d for debuginfo (do not delete after compile)
 %.c: %.d | bin/dpp
 	bin/dpp $< $@
 
diff -urN a/xgcl-2/sysdef.lisp b/xgcl-2/sysdef.lisp
--- a/xgcl-2/sysdef.lisp	2024-01-29 00:10:50.820392725 +0100
+++ b/xgcl-2/sysdef.lisp	2026-08-02 18:29:55.590378395 +0200
@@ -59,7 +59,7 @@
 	(compiler::*default-data-file* t)
 	(compiler::*default-system-p* t))
     (mapc (lambda (x)
-	    (compile-file (format nil "~a.lsp" x) :system-p t))
+	    (compile-file (format nil "~a.lsp" x) :c-file t :h-file t :system-p t))
 	  *files*)))
 
 
